### PR TITLE
TST: Use conda to manage packages in Travis CI

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #403.

Issues with the linux wheels of some dependencies have resulted in incompatibilities in dependent packages in the Travis CI environment.  Moving to a conda install to avoid the pypi wheels.  This allows us to use the latest versions of netcdf4 rather than impose arbitrary caps in the travis environment.

## Type of change

- Testing environment update (non-breaking change which fixes an issue)

# How Has This Been Tested?

Updates Travis CI env, so tested there,.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
develop-3 ==> 1227 passed, 5683 warnings in 1347.17 seconds
this branch ==> 1227 passed, 5682 warnings in 1290.15s

File: .travis.yml [hunk 1]
```diff
@@ -14,19 +14,25 @@ addons:
...
-  - pip install future
-  - pip install pysatCDF >/dev/null
-  # Travis CI currently not working with netCDF4 1.5.3
-  - pip install 'cftime==1.1.1'
-  - pip install 'netCDF4<1.5.3'
-
-# command to install dependencies
 install:
-  - source activate test-environment
+  - sudo apt-get update
```
Comment (aburrell) [inline .travis.yml:18]: Could use some comments here, since this is a lot of commands.

File: .travis.yml [hunk 2]
```diff
@@ -14,19 +14,25 @@ addons:
...
+  - conda info -a
+  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION numpy scipy pandas xarray requests beautifulsoup4 lxml netCDF4 h5py nose
+  - conda activate test-environment
+  - conda install pytest-cov
+  - conda install pytest-ordering
+  - conda install coveralls
+  - conda install future
+  - pip install madrigalWeb
+  - pip install PyForecastTools
+  - pip install pysatCDF >/dev/null
```
Comment (aburrell) [inline .travis.yml:35]: Comment explaining why conda is used for some and pip for others would be useful.

File: pysat/__init__.py [hunk 3]
```diff
@@ -99,7 +99,7 @@
         with open(os.path.join(pysat_dir, 'user_modules.txt'), 'w') as f:
             f.write('')
 
-from pandas import Panel, DataFrame, Series, datetime
+from pandas import DataFrame, Series, datetime
```
Comment (aburrell) [inline pysat/__init__.py:102]: Do we need the datetime here?

Comment (jklenzing) [reply pysat/__init__.py:102]: We need it here until #369 is merged.

**Comment by jklenzing** [comment]
Added comments to travis.yml and tried to simplify.  Thanks!

**Pull request merged**